core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -156,7 +156,8 @@ private[spark] class AppStatusStore(
     // cheaper for disk stores (avoids deserialization).
     val count = {
       Utils.tryWithResource(
-        if (store.isInstanceOf[InMemoryStore]) {
+        if (store.isInstanceOf[ElementTrackingStore]) {
```

Comment:
I think a new method `private def isLiveUI: Boolean` would make the intent here clearer. It can be this instance check or just checking whether `listener` is defined.

Reply:
@vanzin Thanks for the suggestion. I have updated the code.